src/ch04-02-supervision.md

```diff
@@ -28,30 +28,74 @@ The reconfiguration process consists of several stages:
+* `RestartPolicy::on_failure` — actors are restarted only after failures (the `exec` function returned `Err` or panicked).
 * `RestartPolicy::always` — actors are restarted after termination (the `exec` function returned `Ok` or `()`) and failures.
-* `RestartPolicy::never` — actors are never restarted. However, they can be started again on incoming messages.
 
 If the actor is scheduled to be restarted, incoming messages cannot spawn another actor for his key.
 
-Repetitive restarts are limited by a linear backoff mechanism:
-* If there are no restarts for 5s, restart an actor immediately.
-* Otherwise, increase restart time by 5s and schedule restarting.
-* Maximum restart time is limited by 30s.
-
-These constants aren't configurable for now ([elfo#62](https://github.com/elfo-rs/elfo/issues/62)).
 
-The restart policy can be chosen while creating `ActorGroup`:
+The restart policy can be chosen while creating an `ActorGroup`. This is the default restart policy for the group, which will be used if not overridden. 
 ```rust
-use elfo::group::{RestartPolicy, ActorGroup};
+use elfo::{RestartPolicy, RestartParams, ActorGroup};
 
-ActorGroup::new().restart_policy(RestartPolicy::always())
-ActorGroup::new().restart_policy(RestartPolicy::on_failures())
+ActorGroup::new().restart_policy(RestartPolicy::always(RestartParams::new(...)))
+ActorGroup::new().restart_policy(RestartPolicy::on_failure(RestartParams::new(...)))
 ActorGroup::new().restart_policy(RestartPolicy::never())
 ```
 
+The group restart policy can be overridden separately for each actor group in the configuration:
+```toml
+[group_a.system.restart_policy]
+when = "Never" # Override group policy on `config_update` with `RestartPolicy::never()`.
+
+[group_b.system.restart_policy]
+# Override the group policy on `config_update` with `RestartPolicy::always(...)`.
+when = "Always" # or "OnFailure".
+# Configuration details for the RestartParams of the backoff strategy in the restart policy. 
+# The parameters are described further in the chapter.
+min_backoff = "5s" # Required.
+max_backoff = "30s" # Required.
+factor = "2.0" # Optional.
+max_retries = "20" # Optional.
+auto_reset = "5s" # Optional.
+
+[group_c]
+# If the restart policy is not specified, the restart policy from the blueprint will be used.
+```
+
+Additionally, each actor can further override its restart policy through the actor's context:
+```rust
+// Override the group policy with the actor's policy, which is applicable only for this lifecycle.
+ctx.set_restart_policy(RestartPolicy::...);
+
+// Restore the group policy from the configuration (if specified), or use the default group policy.
+ctx.set_restart_policy(None);
+```
+
+### Repetitive restarts
+Repetitive restarts are limited by an exponential backoff strategy. The delay for each subsequent retry is calculated as follows:
+```
+delay = min(min_backoff * pow(factor, power), max_backoff)
+power = power + 1
```

Comment:
I think introducing `power` is useless here. Actually, we're talking about
```
delay = min(min_backoff * pow(factor, restart_count), max_backoff)
          = 0, after resets
```

Reply:
`restart_count`/`restart_count - 1`  would be confusing here